src/Lean/Structure.lean

```diff
@@ -8,25 +8,62 @@ Helper functions for retrieving structure information.
 prelude
 import Lean.Environment
 import Lean.ProjFns
+import Lean.Exception
 
 namespace Lean
 
+/--
+Data for a structure field.
+These are direct fields of a structure, and the full collection of fields is the transitive closure of fields through subobjects.
+-/
 structure StructureFieldInfo where
+  /-- The name of the field. This is a single-component name. -/
   fieldName  : Name
+  /-- The projection function associated to the field. -/
   projFn     : Name
-  /-- It is `some parentStructName` if it is a subobject, and `parentStructName` is the name of the parent structure -/
+  /-- If this field is for a subobject (i.e., an embedded parent structure), contains the name of the parent structure. -/
   subobject? : Option Name
+  /-- The binder info for the field from the `structure` definition. -/
   binderInfo : BinderInfo
+  /-- If set, contains an expression that evaluates to `Syntax` to use as a tactic. From `field := by ...`. -/
   autoParam? : Option Expr := none
   deriving Inhabited, Repr
 
 def StructureFieldInfo.lt (i₁ i₂ : StructureFieldInfo) : Bool :=
   Name.quickLt i₁.fieldName i₂.fieldName
 
-structure StructureInfo where
+/--
+Data for a direct parent of a structure.
+Some structure parents are represented as subobjects (embedded structures),
+and for these the parent project is a true projection.
+Otherwise, the fields of the parent structure are represented among the other fields of the structure,
```

Comment:
Maybe be more explicit, e.g. "If a structure parent shares a field with a previous parent, it will become an implicit parent, and all the fields of the structure parent that do not occur in earlier parents are fields of the new structure".